libs/drape_frontend/relations_draw_info.cpp

```diff
@@ -0,0 +1,83 @@
+#include "relations_draw_info.hpp"
+
+#include "indexer/feature.hpp"
+
+namespace df
+{
+
+void RelationsDrawInfo::Init(FeatureType & ft)
+{
+  using RR = feature::RouteRelationBase;
+
+  buffer_vector<std::pair<std::string, int>, 4> refs;
+  for (uint32_t relID : ft.GetRelations())
+  {
+    auto const rel = ft.ReadRelation(relID);
+    if ((m_sett.hiking && (rel.GetType() == RR::Type::Foot || rel.GetType() == RR::Type::Hiking)) ||
+        (m_sett.cycling && (rel.GetType() == RR::Type::Bicycle || rel.GetType() == RR::Type::MTB)) ||
+        (m_sett.PT &&
+         (rel.GetType() == RR::Type::Bus || rel.GetType() == RR::Type::Tram || rel.GetType() == RR::Type::Trolleybus)))
+    {
+      auto clr = rel.GetColor();
+      if (clr == kEmptyColor)
+        clr = dp::Color{128, 0, 128};  // purple
+
+      // Find an existing color to update frequency.
+      size_t i = 0;
+      size_t const sz = m_colors.size();
+      for (; i < sz; ++i)
+      {
+        if (m_colors[i].first == clr)
+        {
+          m_colors[i].second++;
+          break;
+        }
+      }
+      // Add new color.
+      if (i == sz)
+        m_colors.push_back({clr, 1});
+    }
+
+    if (m_sett.PT)
+    {
+      auto const & r = rel.GetRef();
+      if (!r.empty())
+      {
+        // Get prefix integer value to sort by.
+        char * stop;
+        refs.push_back({r, std::strtol(r.c_str(), &stop, 10)});
```

Comment:
What if strtol fails? It returns 0. Is it ok?
nit:
```suggestion
        refs.emplace_back(r, std::strtol(r.c_str(), &stop, 10));
```